src/align.rs

```diff
@@ -183,6 +192,14 @@ impl<T: Copy, C: AlignedByteChunk> AlignedVec<T, C> {
     }
 
     pub fn resize(&mut self, new_len: usize, value: T) {
+        // In addition to the obvious effect, this verifies the wrapping behavior of the
+        // `new_bytes` calculation. That can not overflow as the length limit does not overflow
+        // when multiplied with the size of `T`. Note that we one can still pass ludicrous
+        // requested buffer lengths, just not unsound ones.
+        assert!(
+            new_len <= Self::MAX_LEN,
```

Comment:
We only use it here and it'd be nicer to see the logic to compute `MAX_LEN` here, so could you move it to this `fn`?

Reply:
Is there any advantage of doing this `<=` check rather than `let new_bytes = new_len.checked_mul(mem::size_of::<T>()).expect("Resizing would overflow the underlying aligned buffer");`?  The latter seems more obviously correct and direct in what it's preventing: overflow.

Reply:
Unfortunately, the constant can not be moved into the function. Rust prevents the local constant from using the outer item's generic parameter. The main advantage relates to the open comment of customizing the panic message. I was unsure if the `MAX_LEN` value should be mentioned for instance. Neither prevents the other, if `checked_mul(..).unwrap()` was used after the manual assert, the bounds check should get removed by the compiler but conversely `expect` can not incorporate the constant in its message.

As an additional consideration, verifying arguments to the function as early as possible. Since this is a precondition rather than an internal logic bug, it might also be helpful to have the constant available for other modules to independently check against some lengths they calculate for future use.

Reply:
I think a normal `.unwrap()`/``.expect()` would probably be fine here as this function shouldn't be very hot.  But we've definitely run into issues where moving the panic and the formatting code it uses to an `#[inline(never)]` (we didn't mark them `#[cold]`, though we probably should) was important, so feel free to do that here as well.

If you'd like to do the `<= MAX_LEN` check first, that's fine, if you pair it with a `.checked_mul(...).unwrap()` that's optimized out so that it's clear which operation exactly we're checking for overflow.  Or you can do the `let Some(_) else` approach you also suggested.